### PR TITLE
Filter out null'd/empty arrays from POST

### PR DESCRIPTION
**Pull request by mjoyce91**
The following are invalid body parameters with the new POST request:
`[null]`
`[]`

This PR ensures that order_by will not be sent like this.

### CODE REVIEW

**Comment by mjoyce91** [comment]
Need to fix so that it works when order_by isn't an array

File: talentmap_api/fsbid/services/available_positions.py [hunk 1]
```diff
@@ -409,7 +409,9 @@ def convert_ap_query(query, allowed_status_codes=["HS", "OP"], isTandem=False, u
         values[f"{prefix}skills2"] = services.convert_multi_value(query.get("position__skill__code__in-tandem"))
     
     if use_post:
-        return pydash.omit_by(values, lambda o: o == None)
+        if isinstance(values[f"{prefix}order_by"], list):
+            values[f"{prefix}order_by"] = pydash.compact(values[f"{prefix}order_by"])
```
Comment (elizabeth-jimenez) [inline talentmap_api/fsbid/services/available_positions.py:413]: Never seen `compact` before; v cool job!

File: talentmap_api/fsbid/services/available_positions.py [hunk 2]
```diff
@@ -409,7 +409,9 @@ def convert_ap_query(query, allowed_status_codes=["HS", "OP"], isTandem=False, u
...
+        return pydash.omit_by(values, lambda o: o is None or (o == [] and bool(o) is False))
```
Comment (elizabeth-jimenez) [inline talentmap_api/fsbid/services/available_positions.py:414]: so i may learn: what's the scenario when `o == [] and bool(o) is True`?

**Review by elizabeth-jimenez — APPROVED** [review]
🍰 👍

File: talentmap_api/fsbid/services/available_positions.py (hunk 2 above)
Comment (mjoyce91) [reply talentmap_api/fsbid/services/available_positions.py:414]: When its an array with a length of >= 1

**Pull request merged**